framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -70,21 +70,14 @@ PatternedMeshGenerator::PatternedMeshGenerator(const InputParameters & parameter
                        " is larger than the the maximum possible index, which is determined by the "
                        "number of MeshGenerators provided in inputs");
 
-  _mesh_ptrs.reserve(_input_names.size());
-  for (auto & input_name : _input_names)
-    _mesh_ptrs.push_back(&getMeshByName(input_name));
-}
-
-std::unique_ptr<MeshBase>
-PatternedMeshGenerator::generate()
-{
   // Reserve spaces for all the meshes
   _meshes.reserve(_input_names.size());
 
   // Read in all of the meshes
   for (MooseIndex(_input_names) i = 0; i < _input_names.size(); ++i)
   {
-    std::unique_ptr<ReplicatedMesh> mesh = dynamic_pointer_cast<ReplicatedMesh>(*_mesh_ptrs[i]);
+    std::unique_ptr<ReplicatedMesh> mesh =
+        dynamic_pointer_cast<ReplicatedMesh>(getMeshByName(_input_names[i]));
```

Comment:
Move this back to the way it was. You can't do this on construction - the mesh generators are not constructed in the order in which they're executed, so the input meshes may not be available yet. 

We can, however, grab a pointer to the unique_ptr up front to see if one with the name exists, but it won't be in a usable state yet.

Reply:
An alternative to this would be to add a method `MooseApp::hasMeshByName(const std::string & name)`, check that up front on the constructor, and then call `getMeshByName()` within `generate()` if you really don't want to store the vector.

Your call.

Reply:
Ok I see the logic. Good catch. I imagine this was the test failures. 
This vector is fine then.